src/AbstractSQLCompiler.ts

```diff
@@ -174,7 +174,12 @@ export type NumberTypeNodes = StrictNumberTypeNodes | UnknownTypeNodes;
 
 export type FieldNode = ['Field', string];
 export type ReferencedFieldNode = ['ReferencedField', string, string];
-export type DateTruncNode = ['DateTrunc', TextTypeNodes, DateTypeNodes];
+export type DateTruncNode = [
+	'DateTrunc',
+	TextTypeNodes,
+	DateTypeNodes,
+	TextTypeNodes?,
+];
```

Comment:
I think this causes `['DateTrunc', TextTypeNodes, DateTypeNodes, undefined]` to be treated as a valid type when it's not, in which case it should be
```suggestion
export type DateTruncNode = [
	'DateTrunc',
	TextTypeNodes,
	DateTypeNodes,
] | [
	'DateTrunc',
	TextTypeNodes,
	DateTypeNodes,
	TextTypeNodes,
];
```
to be more accurate